ios/MeasureSDK/MeasureSDKTests/SessionManagerTests.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  SessionManagerTests.swift
+//  MeasureSDKTests
+//
+//  Created by Adwin Ross on 02/09/24.
+//
+
+import XCTest
+@testable import MeasureSDK
+
+final class SessionManagerTests: XCTestCase {
+    var sessionManager: SessionManager!
+    var idProvider: MockIdProvider!
+    var logger: MockLogger!
+    var configProvider: MockConfigProvider!
+    var randomizer: MockRandomizer!
+    var timeProvider: MockTimeProvider!
+
+    override func setUp() {
+        super.setUp()
+        idProvider = MockIdProvider("test-session-id-1")
+        logger = MockLogger(enabled: false)
+        timeProvider = MockTimeProvider(currentTimeSinceEpochInMillis: 1_000_000_000,
+                                            currentTimeSinceEpochInNanos: 1_000_000_000_000,
+                                            uptimeInMillis: 1_000_000,
+                                            iso8601Timestamp: "2001-09-09T01:46:40.000Z")
+        configProvider = MockConfigProvider(enableLogging: false,
+                                                trackScreenshotOnCrash: false,
+                                                sessionSamplingRate: 1.0,
+                                                eventsBatchingIntervalMs: 1000,
+                                                sessionEndThresholdMs: 1000)
+        randomizer = MockRandomizer(0.5)
+        sessionManager = BaseSessionManager(idProvider: idProvider,
+                                            logger: logger,
+                                            timeProvider: timeProvider,
+                                            configProvider: configProvider,
+                                            randomizer: randomizer)
+    }
+
+    override func tearDown() {
+        sessionManager = nil
+        sessionManager = nil
+        idProvider = nil
+        logger = nil
+        configProvider = nil
+        randomizer = nil
+        timeProvider = nil
+        super.tearDown()
+    }
+
+    func testApplicationWillEnterForeground_DoesNotEndsSession() {
```

Comment:
Consider this name for the test: `testSessionContinues_WhenEnteringForegroundBeforeThreshold`